packages/components/src/validated-form-controls/control-with-error.tsx

```diff
@@ -111,13 +116,58 @@ function UnforwardedControlWithError< C extends React.ReactElement >(
 		};
 	} );
 
-	const validate = () => {
-		const message = customValidator?.();
+	useEffect( () => {
+		if ( ! isTouched ) {
+			return;
+		}
+
 		const validityTarget = getValidityTarget();
 
-		validityTarget?.setCustomValidity( message ?? '' );
-		setErrorMessage( validityTarget?.validationMessage );
-	};
+		if ( ! customValidityMessage?.type ) {
+			validityTarget?.setCustomValidity( '' );
+			setErrorMessage( validityTarget?.validationMessage );
+			setStatusMessage( undefined );
+			return;
+		}
+
+		switch ( customValidityMessage?.type ) {
```

Comment:
The first check is really considering whether `customValidityMessage` is `undefined`, right? Since `type` is expected to always exist if it's defined.

Could we simplify some of these checks to assume if it gets past the first condition, `customValidityMessage` must be truthy.


```suggestion
		if ( ! customValidityMessage ) {
			validityTarget?.setCustomValidity( '' );
			setErrorMessage( validityTarget?.validationMessage );
			setStatusMessage( undefined );
			return;
		}

		switch ( customValidityMessage.type ) {
```

Reply:
The reason I wanted to avoid referencing the `customValidityMessage` object directly in this `useEffect` is so it doesn't need to rerun on object reference changes. But I guess is doesn't matter because `getValidityTarget` is inlined anyway. At the very least we can remove the second conditional though.